_posts/2018-02-12-Reusable-features-in-React-Redux.md

```diff
@@ -0,0 +1,188 @@
+at the end we're trying to solve the same thing. So what is the **reusable feature** ?
+It's basically a standard feature but instead of things defined statically, everything (except components)
+is defined inside **factory functions**. Factory functions allows us to create things in lazy configurable manner.
+Observe e.g. the difference between standard feature actions and reusable one.
+
+
+#### Standard feature actions.js
+
+```js
+const STANDARD_FEATURE_INCREMENT_CLICK = `${namespace}_STANDARD_FEATURE_INCREMENT_CLICK`;
+
+export const incrementClick = createAction(STANDARD_FEATURE_INCREMENT_CLICK);
+```
+     
+#### Reusable feature actions.js
+
+```js
+export default function createActions(namespace, selectors) {
+
+  const REUSABLE_FEATURE_INCREMENT_CLICK = `${namespace}_REUSABLE_FEATURE_INCREMENT_CLICK`;
+
+  const incrementClick = createAction(REUSABLE_FEATURE_INCREMENT_CLICK);
+
+
+  return {
+    REUSABLE_FEATURE_INCREMENT_CLICK: REUSABLE_FEATURE_INCREMENT_CLICK,
+
+    incrementClick,
+  };
+}
+```
+
+Yeah, and that's it. You define everything through the factory functions. 
+At this stage the reusable feature acts as a mere template. Check how the full
+implementation of [reusable feature looks](https://github.com/char0n/react-reusable-features/tree/master/src/app/reusable-feature) in our [reference repository](https://github.com/char0n/react-reusable-features).
+Feature like this cannot be used and mounted to your React application.
+You have to create an **instance** from it.
+
+Creating an instance is as easy as calling the factory functions defined in the reusable feature template.
+Again we adhere to the same rules about directory structure and files names as we did for standard features
+and reusable feature templates. To see how instances are created from the templates please refer 
+the the [reference repository](https://github.com/char0n/react-reusable-features) where we create [instance-1](https://github.com/char0n/react-reusable-features/tree/master/src/app/reusable-feature-instance-1) and [instance-2](https://github.com/char0n/react-reusable-features/tree/master/src/app/reusable-feature-instance-2) from our reusable [feature template](https://github.com/char0n/react-reusable-features/tree/master/src/app/reusable-feature).
+Describing it here in code examples will really be out of scope of this article. 
+Now if you mount these two instances into the React application they work completely independently,
+one doesn't interfere with another, but they share the same code.
+And that's the most important point about reusable features: write once, but use many times...
+ 
+One last note about React+Redux: Even if you are currently using other framework or libraries like Angular or Vue,
+I think it’s important to understand how React and Redux work and how they changed the development of Single Page Applications.
```

Comment:
I noticed this is the first time you are using a smart quote `’` vs `'`. Would you like to update the others to match?